mitodl/ocw-studio
# transcript sync

#### Pre-Flight checklist


#### What are the relevant tickets?
https://github.com/mitodl/ocw-studio/issues/556

#### What's this PR do?
This  PR creates code to download transcripts from 3play using the the 3play api and save them to s3. Links to the s3 files are added to Video objects as new fields.

We will update transcripts when the course is previewed or published. We will also add tasks to update transcripts when the "update" tag is added in the 3play api and to periodically check for transcripts for published courses that don't currently have them.

#### How should this be manually tested?
1) Set  PREPUBLISH_ACTIONS='videos.threeplay_api.update_transcripts_for_website'
Set THREEPLAY_API_KEY to the "api_key" value from 3play/project api key sandbox/3play_sandbox_api_key file in the team keybase
Set THREEPLAY_PROJECT_ID=51817

2) Create a `Video` with a `VideoFile` object with destination `youtube` using the directions here: https://github.com/mitodl/ocw-studio/pull/469

3) Update the destination_id for the the `VideoFile` object created in step 2 to `2gsx-zHwAWs`

4) Publish the website from the UI

5) Check that  'pdf_transcript_s3_url' and  'webvtt_transcript_s3_url' for your video are set.

6) Clear out the data in  'pdf_transcript_s3_url' and  'webvtt_transcript_s3_url' for your video.

7) Go to the 3play UI. Go to the "Sandbox" project. Set the "updated" tag for the video titled "12. Carbohydrates/Introduction to Membranes"

8) Run
```
from videos.tasks import *
update_transcripts_for_updated_videos()
```

9) Check that  'pdf_transcript_s3_url' and  'webvtt_transcript_s3_url' for your video are set again.

10) Clear out the data in  'pdf_transcript_s3_url' and  'webvtt_transcript_s3_url' for your video a second time

11) Run `attempt_to_update_missing_transcripts()`

12) Check that  'pdf_transcript_s3_url' and  'webvtt_transcript_s3_url' for your video are set again.
